packages/wdio-utils/src/envDetector.ts

```diff
@@ -20,30 +20,31 @@ export function isW3C (capabilities?: Capabilities.DesiredCapabilities) {
         return false
     }
 
-    /**
-     * assume session to be a WebDriver session when
-     * - capabilities are returned
-     *   (https://w3c.github.io/webdriver/#dfn-new-sessions)
-     * - it is an Appium session (since Appium is full W3C compliant)
-     */
-    const isAppium = Boolean(
-        capabilities.automationName ||
-        capabilities.deviceName ||
-        capabilities.appiumVersion
-    )
     const hasW3CCaps = Boolean(
         /**
-         * safari docker image may not provide a platformName therefore
-         * check one of the available "platformName" or "browserVersion"
-         */
-        (capabilities.platformName || capabilities.browserVersion) &&
-        /**
-         * local safari and BrowserStack don't provide platformVersion therefore
-         * check also if setWindowRect is provided
+         * Appium can be JWP (Appium 1.x) or W3C (Appium 1.x and 2.x).
+         * JWP for Appium has `platformName` and `platformVersion` as valid
+         * caps which are W3C valid caps for Desktop.
          */
-        (capabilities.platformVersion || Object.prototype.hasOwnProperty.call(capabilities, 'setWindowRect'))
+        isMobile(capabilities) ?
+            /**
+             * The check `isMobile` is not enough to determine if Appium is running with W3C caps
+             */
+            Object.keys(capabilities).find((cap) => cap.startsWith('appium:')) :
```

Comment:
Can we move this into `isMobile` then?

Reply:
I don't think so, because `isMobile` does not determine if it's mobile AND W3C because the starting caps can still be JWP, which in case of Sauce should lead to `tunnelIdentifier` instead of `'sauce:options': {tunnelIdentifier:'tunnelname'}`

Reply:
but if someone applies Appium specific capabilities, we can assume that they want to start a mobile Appium session correct? So `isMobile` should return true?

Reply:
Yes, `isMobile` determines if it's mobile, bot not if it's a W3C session

Reply:
Correct. I think we should generalise this then and check if there are any capabilities that have the format `%s:%s`

Reply:
🤔 , with a generic method it would still not say if it would be a _valid_ Appium W3C session. 
If for example `goog:chromeOptions` or `sauce:options` are used it's still not a valid Appium W3C session. A valid one would be **with** `appium:%s`, if they provide other wrong options which should have been in the format `%s:%s` then it's the driver/vendor who should fail

Reply:
The goal of this method (`isW3C`) is to find out whether or not the session is a W3C session or not, no matter if run on mobile or desktop. My suggestion was to check if any of the keys within the session response have the following format: `%s:%s`. If one of them has this format we have to assume W3C compatibility.

Reply:
Also can we format this to be:
```js
const hasW3CCaps = <condition>
    ? <value if true>
    : <value if false>
```
makes it easier to read IMO. I was overseeing the `:` at end of that one line.

Reply:
@christian-bromann 

I've checked the session response data of a view mobile sessions and also from desktop and I didn't see any `%s:%s` coming back. Should this be returned in a W3C response?